client/grpc/tmservice/service.go

```diff
@@ -78,6 +81,26 @@ func (s queryServer) GetBlockByHeight(ctx context.Context, req *GetBlockByHeight
 	if err != nil {
 		return nil, err
 	}
+
+	var txns []*txtypes.Tx
+	if res.Block != nil {
+		txns = make([]*txtypes.Tx, len(res.Block.Txs))
+		for i, tx := range res.Block.Data.Txs {
+			txHexStr := fmt.Sprintf("%X", tmhash.Sum(tx))
+			res, err := authtx.QueryTx(s.clientCtx, txHexStr)
```

Comment:
Why are we querying for the tx here? IIRC, the `txs` slice in the Tendermint response includes base64 encoded tx blobs. I.e. you base64 decode, and then take those bytes and decode them into a tx.

Reply:
ok, switched to decoding these to Any's and placed that in the response. I did fiddle around with seeing if i could get it back to our protobuf Tx type, but no dice-- let me know if this is okay as is?